Working on constructor

paulaOliverosHollis opened the pull request:
Added all the file stuff!

Commit:
Now we check if there is a score file or not in the constructor. If it exists, we read from it into a list and if it does not exist we create it.

Also added exeption catching.

The reading from the file is handle by its own function.

zhollis21 commented:
@paulaOliverosHollis You will need to fix the conflicts by getting the latest master and then merging master into this branch

Commit:
Updated this branch with the latest from master.

Commit:
Making sure I have the latest from master

zhollis21 commented on DeveloperGame/DeveloperGame/ComputerGeekGame.cs:
```diff
@@ -1,10 +1,30 @@
 public class ComputerGeekGame
 {
+    private string _scoreBoardfile = "DeveloperGameLeaderBoard.txt";
+    private List<KeyValuePair<int, string>> _leaderBoard = new List<KeyValuePair<int, string>>();
```
If you want to store the Name, Score, and Time, this won't work as it can only hold two things. You will probably need to create a custom class to hold all the data.

paulaOliverosHollis replied:
Should I just drop the idea of the time?

zhollis21 replied:
No, keep it just make another class that holds all 3 fields

paulaOliverosHollis replied:
Fixed.

zhollis21 commented on DeveloperGame/DeveloperGame/ComputerGeekGame.cs:
```diff
@@ -1,10 +1,30 @@
 
 public class ComputerGeekGame
 {
+    private string _scoreBoardfile = "DeveloperGameLeaderBoard.txt";
```
Should the file name be: "ComputerGeekLeaderBoard"?

Also because this goes in the same folder as the .exe, I think it would be sufficient to just say "leaderboard.txt", unless you plan on having multiple different types of leaderboards...

zhollis21 replied:
Also your variable name would imply that it is a file, but it is only string, so I would say "filename" instead of just "file"

paulaOliverosHollis replied:
Fixed.

zhollis21 commented on DeveloperGame/DeveloperGame/ComputerGeekGame.cs:
```diff
@@ -25,8 +45,37 @@ public void Run()
+        {
+            string[] scoreAndName = line.Split(' ');
+
+            if (scoreAndName.Length == 2)
```
Won't you have 3 things? Name, Score and Time?

paulaOliverosHollis replied:
Yes, I did not really think through how I am going to get the time in there.

paulaOliverosHollis replied:
Fixed.

zhollis21 commented on DeveloperGame/DeveloperGame/ComputerGeekGame.cs:
```diff
@@ -25,8 +45,37 @@ public void Run()
+
+            if (scoreAndName.Length == 2)
+            {
+                bool isItConvertible = int.TryParse(scoreAndName[0], out int result);
```
Also you don't have to do this, but I would recommend putting the name first, then score, then time. I think this is the convention for games

paulaOliverosHollis replied:
Ok, sounds good.

paulaOliverosHollis replied:
Fixed.

zhollis21 commented on DeveloperGame/DeveloperGame/ComputerGeekGame.cs:
```diff
@@ -1,10 +1,30 @@
+            {
+                FileStream stream = File.Create(_scoreBoardfile);
+                stream.Close();
+            }
```
Are you going to create some placeholder scores if they are playing for the first time and don't yet have a score board? Or if there was nothing in the file?

paulaOliverosHollis replied:
I am not planning on creating any place holder scores.

zhollis21 replied:
Ok, just keep that in mind they might be empty when using the list of scores

zhollis21 commented on DeveloperGame/DeveloperGame/ComputerGeekGame.cs:
```diff
@@ -25,8 +45,37 @@ public void Run()
             }
-           
-        } while (!UserChoseToQuit(userMenuChoice));
+            if (UserChoseToQuit(userMenuChoice))
```
your do-while loop got messed up.

paulaOliverosHollis replied:
Fixed.

zhollis21 commented on DeveloperGame/DeveloperGame/ComputerGeekGame.cs:
```diff
@@ -1,10 +1,30 @@
+        }
+        catch (Exception e)
+        {
+            Console.WriteLine($"\nSomething went wrong. The following error was generated: {e.Message} Please exit the game and try again.");
```
Also, it might be helpful to specify when it went wrong... ex: "Something went wrong when trying to load the leaderboards."

paulaOliverosHollis replied:
Fixed.

Commit:
Created a new custom class that holds the player's name, points and time.

When a file exists we know read  all the scores into a lists of Scores instead of keyValue pairs.

Wrapped everything in the same namespace and fix some small issues from the pull request.

zhollis21 commented on DeveloperGame/DeveloperGame/ComputerGeekGame.cs:
```diff
@@ -1,98 +1,151 @@
-            PrintMenu();
-            userMenuChoice = Console.ReadLine();
+            Console.WriteLine("\nWelcome To Computer Geek Game!");
+            Console.WriteLine("\nIn this game, your knowledge about Computer Science history is going to be tested. The test cosists of multiple choice and true or false questions. " +
```
This is just a suggestion and can be fixed in the next pull request, but instead of doing two writelines(), just call it once and use new lines.

zhollis21 commented on DeveloperGame/DeveloperGame/Score.cs:
```diff
@@ -0,0 +1,56 @@
+        private int _points;
+        private TimeSpan _time;
+        
+        public string PlayerName
```
Unless you are going to do something special in your getter or setter, just turn this into an auto-property

![image](https://user-images.githubusercontent.com/43797367/62795807-8d081c80-baa5-11e9-8ac3-c3385227c184.png)

zhollis21 reviewed (approved):
Looks good, just fix those small things in another pull request.
